sdk/storage/Azure.Storage.DataMovement/src/LocalTransferCheckpointer.cs

```diff
@@ -153,19 +153,22 @@ public override async Task<Stream> ReadJobPlanFileAsync(
             CancellationHelper.ThrowIfCancellationRequested(cancellationToken);
             if (_transferStates.TryGetValue(transferId, out JobPlanFile jobPlanFile))
             {
-                // Lock MMF
                 await jobPlanFile.WriteLock.WaitAsync().ConfigureAwait(false);
+                try
+                {
+                    using (MemoryMappedFile mmf = MemoryMappedFile.CreateFromFile(jobPlanFile.FilePath))
+                    using (MemoryMappedViewStream mmfStream = mmf.CreateViewStream(offset, length, MemoryMappedFileAccess.Read))
+                    {
+                        await mmfStream.CopyToAsync(copiedStream).ConfigureAwait(false);
+                    }
 
-                using (MemoryMappedFile mmf = MemoryMappedFile.CreateFromFile(jobPlanFile.FilePath))
-                using (MemoryMappedViewStream mmfStream = mmf.CreateViewStream(offset, length, MemoryMappedFileAccess.Read))
+                    copiedStream.Position = 0;
+                    return copiedStream;
```

Comment:
Is this not supposed to release the write lock after it returns the stream here? Or does the lock release when it returns the stream.